cli/js/net.ts

```diff
@@ -123,6 +148,50 @@ export class ListenerImpl implements Listener {
   }
 }
 
+export class UDPConnImpl implements UDPConn {
+  constructor(
+    readonly rid: number,
+    readonly addr: Addr,
+    public bufSize: number = 1024,
+    private closing: boolean = false
+  ) {}
+
+  async receive(): Promise<[Uint8Array, Addr]> {
+    const p = new Uint8Array(this.bufSize);
```

Comment:
The user should have the ability to supply a Uint8Array to recv data into. So I think this method should take an argument Uint8Array...

Reply:
This is not compatible with the iterator properties of UDPConn, we could not do the following:
    
    for await (const msg of socket){
        // ...
    }

Reply:
Hm - I'm sure it could be made to work somehow. I can have a look later.

Reply:
I made it optional, creating a new buffer in no one is passed